Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -73,14 +73,26 @@ struct SymbolGraphLoader {
                 }
 
                 // `moduleNameFor(_:at:)` is static because it's pure function.
-                let (moduleName, _) = Self.moduleNameFor(symbolGraph, at: symbolGraphURL)
+                let (moduleName, isMainSymbolGraph) = Self.moduleNameFor(symbolGraph, at: symbolGraphURL)
                 // If the bundle provides availability defaults add symbol availability data.
                 self.addDefaultAvailability(to: &symbolGraph, moduleName: moduleName)
 
+                // main symbol graphs are ambiguous
+                var usesExtensionSymbolFormat: Bool? = nil
+                
+                // transform extension block based structure emitted by the compiler to a
+                // custom structure where all extensions to the same type are collected in
+                // one extended type symbol
+                if !isMainSymbolGraph {
```

Comment:
That's not quite correct. The feature flag in swiftSymbolGraphGen enables the emission of `swift.extension` symbols. These, however, only appear in "extension symbol graph files", i.e. those where the name follows the `ExtendingModule@ExtendedModule.symbols.json` pattern. The condition you point out here only makes sure we don't apply the transformation on main symbol graph files, as these can't contain `swift.extension` symbols anyway. 

The transformation scans the graph's symbols for such of kind `swift.extension`. It there are no such symbols, it aborts and reports `containsExtensionSymbols = false`. Otherwise, the transformation is applied. That is, the presence of `swift.extension` symbols in extension symbol graph files activates the extension support feature in Swift-DocC.

Therefore, the feature flag on the Swift compiler is enough, as Swift-DocC automatically adapts its behavior to the output.

What's important to understand is that the later portions of Swift-DocC can't deal with `swift.extension` symbols, but only with the symbols and structure generated by the transformation, i.e. the symbols `swift.module.extension`, `swift.struct.extension`, ..., `swift.unknown.extension`.

Thus, adding a feature flag to Swift-DocC could only introduce a situation where the execution of Swift-DocC fails because it expects a different input format.

> Workflows may be relying on the current behavior for what happens when passing extension symbol graph files to DocC.

Yes, and they will continue to get the current behavior as long as they don't enable the extension feature on the Swift compiler (or we make it the default), because only the presence of `swift.extension` symbols activates the new behavior.

Reply:
Thanks for this extra explanation. 

Now that I've gone through `ExtendedTypesFormatTransformation` code in detail and understand the extension symbols this makes sense to me. I agree that DocC doesn't need another feature flag for this.